enrollment-server/src/main/java/com/wultra/app/enrollmentserver/impl/service/MobileTokenService.java

```diff
@@ -200,6 +205,10 @@ public void operationFailApprove(@NotNull String operationId, @NotNull RequestCo
         // Prepare additional data
         request.getAdditionalData().put(ATTR_IP_ADDRESS, requestContext.getIpAddress());
         request.getAdditionalData().put(ATTR_USER_AGENT, requestContext.getUserAgent());
+        final UserAgent.Device device = UserAgent.parse(requestContext.getUserAgent());
```

Comment:
Out of the scope of this PR, but `UserAgent.parse` is missing Javadoc, and instead of `null` it could return `Optional`.

Reply:
See https://github.com/wultra/lime-java-core/pull/230